### ISSUE TITLE
amp-youtube placeholder image resized incorrectly (stretched)

### ISSUE THREAD

**Issue by sryze** [ampproject/amphtml#22836]
## What's the issue?

If I put a placeholder image inside of `<amp-youtube>` as shown [here](https://amp.dev/documentation/components/amp-youtube) and use a responsive layout instead of fill the image is stretched to fill the container nevertheless (as if the layout attribute is ignored).

However, if I remove the `placeholder` attribute, the image is not stretched, but obviously it does not work as a placeholder and AMP tries to load the default thumbnail image for the video.

I'm trying to use this to replace the default `sddefault.jpg` image with `mqdefault.jpg` or `hqdefault.jpg` because `sddefault.jpg` doesn't exist for my video (and I can't reupload the video). If there is a simpler way to do this feel free to let me know 👍 

(code from that link for reference:

```
<amp-youtube
      id="myLiveChannel"
      data-live-channelid="UCB8Kb4pxYzsDsHxzBfnid4Q"
      width="358"
      height="204"
      layout="responsive">
    <amp-img
      src="https://i.ytimg.com/vi/Wm1fWz-7nLQ/hqdefault_live.jpg"
      placeholder
      layout="fill"
      />
  </amp-youtube>
```
)

## How do we reproduce the issue?

See this page: 

https://sryze.github.io/amp-youtube-placeholder-bug/index.html

When you open it you will see the buggy image for a couple of seconds, then AMP will display the YouTube iframe.

## What browsers are affected?

Tested in Chrome, Firefox

## Which AMP version is affected?

1906131851520

**Comment by aghassemi** [comment]
Hi @sryze 

The aspect ratio of the image is different than the aspect ratio of the video. You would want to add `object-fit="cover"` to the `amp-img` to solve the issue.

**Comment by sryze** [comment]
But why does it work perfectly fine without "placeholder"? seems like a bug with placeholders to me

**Comment by aghassemi** [comment]
@sryze when we put the default placeholder ourselves, we apply the `object-fit:cover` as well. If you are customizing the placeholder, then it is up to you, you may want to do `object-fit:container` for instance, we don't make assumptions in that case.

**Comment by sryze** [comment]
Thanks for clarifying. I'll try using that property.

**Comment by sryze** [comment]
Added this to my style sheet:

```
amp-youtube amp-img[placeholder] img {
    object-fit: cover;
}
```

and it's working (though not in IE 11).
